### PR TITLE
Remove failure analysis and apply code cleanup

### PR DESCRIPTION
**Pull request by hallipr**

### CODE REVIEW

File: tools/pipeline-witness/infrastructure/Assign-StoragePermissions.ps1 [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    Write-Error "Failed to grant access"
+    exit 1
+  }
+    
+  Write-Host "Granting 'Storage Queue Data Contributor' access to $appResourceGroupName/$appStorageAccountName"
+  $scope = "/subscriptions/$subscriptionId/resourceGroups/$appResourceGroupName/providers/Microsoft.Storage/storageAccounts/$appStorageAccountName"
+  $output = Invoke "az role assignment create --assignee '$azAdGroupId' --role 'Storage Queue Data Contributor' --scope '$scope' --output none"
+    
+  if ($LASTEXITCODE -ne 0) {
+    Write-Output $output
```
Comment (benbp) [inline tools/pipeline-witness/infrastructure/Assign-StoragePermissions.ps1:63]: I'm ok with this but is there a reason you didn't put the role assignments into the bicep templates?

**Review by benbp — APPROVED** [review]

File: tools/pipeline-witness/infrastructure/Assign-StoragePermissions.ps1 (hunk 1 above)
Comment (hallipr) [reply tools/pipeline-witness/infrastructure/Assign-StoragePermissions.ps1:63]: The storage assignments are only needed when a dev is running the service locally and connecting to live resources. This doesn't need to be persistent access

Comment (hallipr) [reply tools/pipeline-witness/infrastructure/Assign-StoragePermissions.ps1:63]: Also, in the pipeline, the deployment principal doesn't have permission to query Entra.  We'd need to hard code the group's object id into the parameters and that didn't seem correct.

**Pull request merged**